decidim-core/app/services/decidim/truncation.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module Decidim
+  class Truncation
+    include ActionView::Context
+    include ActionView::Helpers::TagHelper
+
+    # Truncates text and html content
+    # text - Content to be truncated
+    # options - Hash with the options
+    #         max_length: An Integer maximum number of characters
+    #         tail: A string suffix to be added after truncation
+    #         count_tags: A boolean if html is calculate to max length, otherwise just content
+    #         count_tail: A boolean value that determines whether max_length contains the tail
+    #         tail_before_final_tag: A boolean, show tail inside of tag where text is cutted or before final closing tag.
+    def initialize(text, options = {})
+      @options = {
+        max_length: options[:max_length] || 30,
+        tail: options[:tail] || "...",
+        count_tags: options[:count_tags] || false,
+        count_tail: options[:count_tail] || false,
+        tail_before_final_tag: options[:tail_before_final_tag] || false
+      }
+      @document = Nokogiri::HTML::DocumentFragment.parse(text)
+      @tail_added = false
```

Comment:
I think the state variables should be initialized in the `truncate` method so that it would also work when `truncate` is called multiple times consecutively.